src/ha-utils/ha/gcp/providers/GCPIAAS.go

```diff
@@ -0,0 +1,361 @@
+		modifyLBInput.Fingerprint = loadBalancer.Fingerprint
+		modifyLBInput.ConnectionDraining = connDrain
+
+		if loadBalancer.Backends == nil || len(loadBalancer.Backends) <= 0 {
+			isUpdateRequired = true
+		} else {
+			for _, currBackend := range loadBalancer.Backends {
+				isUpdateRequired = true
+				if strings.Compare(strings.TrimSpace(currBackend.Group), strings.TrimSpace(vmGroup.SelfLink)) == 0 {
+					log.Println("This VM group is already present in load balancer. IG: ", currBackend.Group)
+					isUpdateRequired = false
+					break
+				}
+			}
+		}
+		if isUpdateRequired == true {
+			modifyLBInput.Backends = append(loadBalancer.Backends, modifyBackend)
+			returnValue = iaasProvider.lbService.UpdateLoadBalancer(modifyLBInput, regionName)
+			if returnValue == false {
+				return 11
+			}
+			log.Println("Load balancer with name ", loadBalancerName, "updated successfully.")
+			loadBalancer, returnValue = iaasProvider.lbService.GetLoadBalancer(loadBalancerName, regionName)
+			if returnValue == false {
+				return 12
+			}
+		}
+	}
+	// If the load balancer doesn't exist, then create one.
+	if loadBalancer == nil {
+		log.Println("Load balancer with name ", loadBalancerName, "does not exist in GCP. Initiating creation of load balancer resource.")
+
+		var createLBInput gcpmodels.CreateLBInput = gcpmodels.CreateLBInput{}
+		var connDrain gcpmodels.ConnectionDrainInfo = gcpmodels.ConnectionDrainInfo{30}
+		var backend gcpmodels.Backend = gcpmodels.Backend{}
+		backend.BalancingMode = "CONNECTION"
+		backend.Group = vmGroup.SelfLink
+		
+		createLBInput.Name = loadBalancerName
+		createLBInput.LoadBalancingScheme = "INTERNAL"
+		createLBInput.Protocol = "TCP"
+		createLBInput.HealthChecks = []string{probe.SelfLink}
+		createLBInput.Backends = []gcpmodels.Backend{backend}
+		createLBInput.ConnectionDraining = connDrain
+		returnValue = iaasProvider.lbService.CreateLoadBalancer(createLBInput, regionName)
+		if returnValue == false {
+			return 13
+		}
+		loadBalancer, returnValue = iaasProvider.lbService.GetLoadBalancer(loadBalancerName, regionName)
+		if returnValue == false {
+			return 14
+		}
+	}
+
+	// Identify whether a load balancing rule exists with the given name.
+	lbRuleName = iaasProvider.getLoadBalancingRuleName()
+	lbRule, returnValue = iaasProvider.lbRuleService.GetLBRule(lbRuleName, regionName)
+	if returnValue == false {
+		log.Println("Failed to retrieve load balancing rule details - ", loadBalancerName)
+		return 15
+	} else if returnValue == true && lbRule != nil {
+		log.Println("Load balancing rule with name:", lbRuleName, "already exists in GCP.")
+	}
+	// If the load balancing rule doesn't exist, then create one.
+	if lbRule == nil {
+		log.Println("Load balancing rule with name ", lbRuleName, "does not exist in GCP. Initiating creation of load balancing rule resource.")
+
+		floatingIPNetwork, floatingIPSubNetwork, returnValue := iaasProvider.getNetworkInfo(iaasProvider.Config.SubnetName, regionName)
```

Comment:
`getNetworkInfo` can return empty and false values. Dont see any checks being done here if this method succedeed before proceeding. Should you be returning failure here if this method failed?